src/main/java/com/synopsys/integration/alert/channel/azure/boards/web/AzureBoardsCustomEndpoint.java

```diff
@@ -22,37 +22,118 @@
+    public AzureBoardsCustomEndpoint(CustomEndpointManager customEndpointManager, ResponseFactory responseFactory, Gson gson, AlertProperties alertProperties, ConfigurationAccessor configurationAccessor,
+        ConfigurationFieldModelConverter modelConverter)
+        throws AlertException {
+        super(AzureBoardsDescriptor.KEY_OAUTH, customEndpointManager, responseFactory, gson);
+        this.alertProperties = alertProperties;
+        this.configurationAccessor = configurationAccessor;
+        this.modelConverter = modelConverter;
     }
 
     @Override
-    public ResponseEntity<String> createResponse(FieldModel fieldModel, HttpServletContentWrapper servletContentWrapper) {
-        return responseFactory.createOkResponse("", "Placeholder Message"); //FIXME this message should fixed once the Oauth is implemented.
+    protected OAuthEndpointResponse createOAuthResponse(FieldModel fieldModel, HttpServletContentWrapper servletContentWrapper) {
+        try {
+            FieldAccessor fieldAccessor = createFieldAccessor(fieldModel);
+            Optional<String> clientId = fieldAccessor.getString(AzureBoardsDescriptor.KEY_CLIENT_ID);
+            if (!clientId.isPresent()) {
+                return new OAuthEndpointResponse(HttpStatus.BAD_REQUEST.value(), false, "", "client id not found.");
+            }
+            Optional<String> alertServerUrl = alertProperties.getServerUrl();
+
+            if (!alertServerUrl.isPresent()) {
+                return new OAuthEndpointResponse(HttpStatus.BAD_REQUEST.value(), false, "", "Could not determine the alert server url for the callback.");
+            }
+            String authUrl = createAuthURL(clientId.get(), alertServerUrl.get());
+            logger.info("Authenticating Azure OAuth URL: " + authUrl);
+            //TODO add code to check if Alert has already been authorized to set the authenticated flag.
+
+            return new OAuthEndpointResponse(HttpStatus.OK.value(), false, authUrl, "");
+
+        } catch (Exception ex) {
+            logger.error("Error activating Azure Boards", ex);
+            return new OAuthEndpointResponse(HttpStatus.INTERNAL_SERVER_ERROR.value(), false, "", "Error activating azure oauth.");
+        }
+    }
+
+    private FieldAccessor createFieldAccessor(FieldModel fieldModel) {
+        Map<String, ConfigurationFieldModel> fields = new HashMap<>();
+        try {
+            fields.putAll(modelConverter.convertToConfigurationFieldModelMap(fieldModel));
+            // check if a configuration exists because the client id is a sensitive field and won't have a value in the field model if updating.
+            if (StringUtils.isNotBlank(fieldModel.getId())) {
+                Optional<ConfigurationModel> configurationFieldModel = configurationAccessor.getConfigurationById(Long.valueOf(fieldModel.getId()));
+                configurationFieldModel.ifPresent(model -> fields.putAll(model.getCopyOfKeyToFieldMap()));
```

Comment:
Could we do something like this?
```suggestion
                configurationAccessor.getConfigurationById(Long.valueOf(fieldModel.getId()))
                .map(ConfigurationModel::getCopyOfKeyToFieldMap)
                .ifPresent(fields::putAll);
```